File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.py
def solution(record):
    result = []
    nameDic = {}
    answer=[]
    for i in record:
        temp=i.split(' ')
        if temp[0]=='Enter' or temp[0]=="Change":
            nameDic[temp[1]]=temp[2]
        result.append(temp)

    for i in result:
        if i[0] =='Enter':
            answer.append(f'{nameDic[i[1]]}님이 들어왔습니다.')
        elif i[0] == 'Leave':
            answer.append(f'{nameDic[i[1]]}님이 나갔습니다.')

    return answer

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.js
function solution(record) {
    const nameMap = new Map()
    const temp=record.map((v)=>v.split(' '))
    
    temp.forEach((v)=>{
        if(v[0]==='Enter' || v[0]==='Change'){
            nameMap.set(v[1],v[2])
            
        }
        
    })
    const answer=[]
    temp.forEach((v)=>{
        if(v[0]==='Enter'){
            answer.push(`${nameMap.get(v[1])}님이 들어왔습니다.`)
        }
        if(v[0]==='Leave'){
            answer.push(`${nameMap.get(v[1])}님이 나갔습니다.`)
        }
        
    })
    return answer
    
}